### PR TITLE
Db models

### PR DESCRIPTION
**Pull request by manuelsaezcarmona**

### CODE REVIEW

**Review by manuelsaezcarmona — COMMENTED** [review]
Perfect

File: controllers/user.controller.js [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
-        msg: 'Ya existe un usuario con ese correo'
+        msg: 'Ya existe un usuario con ese correo',
       });
     }
-    const newUser = new User(user);
+    const userContent = { username, email, password };
+    userContent.posts = [];
+    userContent.comments = [];
+
+    const newUser = new User(userContent);
```
Comment (jvillarreal-ut) [inline controllers/user.controller.js:22]: Check suggestion works, so you can take advantage of destructuring
`const newUser = new User({
  username,
  email,
  password
});`

File: controllers/user.controller.js [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
+    newUser.password = bcrypt.hashSync(userContent.password, salt);
 
     await newUser.save();
+
     return res.status(201).json({
       ok: true,
-      msg: 'usuario creado',
-      username: user.username
+      msg: 'create user',
+      username: newUser.username,
```
Comment (jvillarreal-ut) [inline controllers/user.controller.js:32]: Take advantage of destructuring. You can define it using only `username`

File: models/comment.model.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const { Schema, model } = require('mongoose');
+
+const CommentSchema = Schema({
+  commenttext: {
```
Comment (jvillarreal-ut) [inline models/comment.model.js:4]: Suggestion: rename as `content`

File: models/post.model.js [hunk 4]
```diff
@@ -3,33 +3,23 @@ const { Schema, model } = require('mongoose');
 const PostSchema = Schema({
   imagepost: {
```
Comment (jvillarreal-ut) [inline models/post.model.js:4]: Suggestion: rename as `imageUrl`

File: models/post.model.js [hunk 5]
```diff
@@ -3,33 +3,23 @@ const { Schema, model } = require('mongoose');
...
     type: String,
-    required: true
+    required: true,
   },
   textpost: {
```
Comment (jvillarreal-ut) [inline models/post.model.js:8]: Suggestion: rename as `content`

File: routes/user.routes.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
-const router = require('express').Router();
+const usersRouter = require('express').Router();
 const { addUser } = require('../controllers/user.controller');
 
-router.post('/new', addUser);
+usersRouter.post('/new', addUser);
```
Comment (jvillarreal-ut) [inline routes/user.routes.js:4]: Suggestion: following REST pattern, it should be interesting to remove `new` subpath, as far as u are using `POST` action

**Pull request merged**